controller/system_backup_controller.go

```diff
@@ -268,13 +269,16 @@ func (c *SystemBackupController) syncSystemBackup(key string) (err error) {
 	}
 
 	backupTarget, err := c.ds.GetDefaultBackupTargetRO()
-	if err != nil {
+	if err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
 
-	backupTargetClient, err := newBackupTargetClientFromDefaultEngineImage(c.ds, backupTarget)
-	if err != nil {
-		return err
+	var backupTargetClient *engineapi.BackupTargetClient
+	if backupTarget != nil {
```

Comment:
If `backupTarget` is nil, it should just return and wait for the next reconcile instead. Otherwise, the following `c.reconcile(..)` might hit a nil exception when accessing `c.backupTargetClient` or `backupTarget`.

Reply:
The default backup target will be nil (deleted before system backups are deleted ) only when uninstalling.
If we just return and wait for the next reconcile, the system backup will never clean up the finalizer when uninstalling.

>Otherwise, the following c.reconcile(..) might hit a nil exception when accessing c.backupTargetClient or backupTarget.

The system backups not in the final state (`ready`, `error`, and `deleting`) will be updated to the state `error` if the backup target is `nil` (the backup target does not exist.), and `case longhorn.SystemBackupStateReady, longhorn.SystemBackupStateError:` will not use the `c.backupTargetClient` or `backupTarget`.